src/openrct2/peep/Peep.cpp

```diff
@@ -3271,6 +3271,17 @@ void rct_peep::PerformNextAction(uint8 & pathing_result, rct_tile_element * & ti
                 return;
             }
 
+            if (type == PEEP_TYPE_STAFF && !GetNextIsSurface())
+            {
+                // Prevent staff from leaving the path on their own unless they're allowed to mow.
+                if (!((this->staff_orders & STAFF_ORDERS_MOWING) && this->staff_mowing_timeout >= 12))
```

Comment:
The check for `staff_mowing_timeout` seems unnecessary here. Withing the 12 ticks, the staff will not be able to get on a path and try to leave it, and even if they do it's only preventing them from leaving the path while they should be able to go off of it.

Reply:
The same check is used to determine where it should go to mow the grass. So I'm using it here as well since the staff should only be able to walk off a path if it is ready to mow.